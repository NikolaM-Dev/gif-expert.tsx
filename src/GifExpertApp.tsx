interface IProps {}

export const GifExpertApp = ({}: IProps): JSX.Element => {
  return (
    <>
      <h1>GifExpertApp</h1>
    </>
  );
};
