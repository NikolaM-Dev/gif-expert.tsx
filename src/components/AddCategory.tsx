import { FormEvent, SyntheticEvent, useState } from 'react';

interface IProps {
  handleAddCategory: (category: string) => void;
}

export const AddCategory = ({ handleAddCategory }: IProps) => {
  const [newCategory, setNewCategory] = useState<string>('');

  const handleChange = ({ currentTarget }: FormEvent<HTMLInputElement>) =>
    setNewCategory(currentTarget.value);

  const handleSubmit = (e: SyntheticEvent) => {
    e.preventDefault();

    handleAddCategory(newCategory);
    setNewCategory('');
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        onChange={handleChange}
        placeholder="Search Gifs"
        type="text"
        value={newCategory}
      />
    </form>
  );
};
