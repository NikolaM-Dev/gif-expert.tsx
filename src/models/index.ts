export * from './ghipy-response.model';
export * from './gif.model';
