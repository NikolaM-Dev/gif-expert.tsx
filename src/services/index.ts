export * from './get-gifs.service';
