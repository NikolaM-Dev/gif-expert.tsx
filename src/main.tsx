import React from 'react';
import { createRoot } from 'react-dom/client';

import { GifExpertApp } from './GifExpertApp';

import './styles.css';

createRoot(document.getElementById('root') as HTMLElement).render(
  <React.StrictMode>
    <GifExpertApp />
  </React.StrictMode>,
);
