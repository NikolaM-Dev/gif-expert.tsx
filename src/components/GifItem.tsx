import { IGif } from '../models';

interface IProps {
  gif: IGif;
}

export const GifItem = ({ gif }: IProps): JSX.Element => {
  return (
    <article className="card">
      <img src={gif.url} alt={gif.title} />
      <p>{gif.title}</p>
    </article>
  );
};
