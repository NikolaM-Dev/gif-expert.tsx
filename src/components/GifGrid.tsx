import { GifItem } from './GifItem';
import { IGif } from '../models';
import { useFetchGifs } from '../hooks';

interface IProps {
  category: string;
}

export const GifGrid = ({ category }: IProps): JSX.Element => {
  const { gifs, isLoading } = useFetchGifs(category);

  return (
    <>
      <h2>{category}</h2>

      {isLoading && <p>Is Loading...</p>}

      <section className="card-grid">
        {gifs.map((gif: IGif) => (
          <GifItem key={gif.id} gif={gif} />
        ))}
      </section>
    </>
  );
};
