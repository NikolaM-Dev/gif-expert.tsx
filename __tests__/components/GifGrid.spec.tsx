import { render, screen } from '@testing-library/react';

import { GifGrid } from '../../src/components';

describe('Tests on <GifGrid />', () => {
  const category = 'Dragon Ball Z';

  it('should to match with snapshot', () => {
    const { container } = render(<GifGrid category={category} />);

    expect(container).toMatchSnapshot();
  });
});
