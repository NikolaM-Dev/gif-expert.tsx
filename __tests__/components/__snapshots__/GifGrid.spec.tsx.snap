// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGrid /> should to match with snapshot 1`] = `
<div>
  <h2>
    Dragon Ball Z
  </h2>
  <p>
    Is Loading...
  </p>
  <section
    class="card-grid"
  />
</div>
`;
